firehose.tf

```diff
@@ -0,0 +1,38 @@
+resource "aws_s3_bucket" "firehose_bucket" {
+  count       = var.firehose_enabled ? 1 : 0
+  bucket      = module.this.id
```

Comment:
this bucker name could collide with some other buckets we'd want to provision.
I'd use a separate `label` with `attributes`

```
module "s3_firehose_label" {
  source  = "cloudposse/label/null"
  version = "0.24.1"

  attributes = ["firehose"]

  context = module.this.context
```

```
bucket      = module.s3_firehose_label.id
```

it solves two potential issues:

1. name collision
2. looking at the bucket, it would be easier to understand that the bucket is related to `firehose`